### PR TITLE
D3ASIM-3653: Added correct trade event triggering in the LoadHours st…

### PR DESCRIPTION
**Pull request by hannesdiedrich**
…rategy.

`settlement_market_strategy.event_bid_traded` was not correctly triggered in the LoadHours strategy due to an early return of the calling method (`event_bid_traded`) 
and 
`settlement_market_strategy.event_trade` where not correctly triggered (instead `event_bid_traded` was triggered before).

I also added an integrationtest. Unfortunately, I could not reuse an existing setup file and had to use a stripped down version of the one Andrea provided when posting this issue.

### CODE REVIEW

**Review by fievelk — APPROVED** [review]
LGTM.

Nit. We currently have two events whose name and implementations seem a bit odd to me: 
- `MarketEvent.BID_TRADED` for bids
- `MarketEvent.TRADE` for offers

This reflects in the methods `event_bid_traded()` and `event_traded()`, which are basically doing the same thing but one with bids and the other with offers. I would personally either unify them or rename the offer event to `MarketEvent.OFFER_TRADED`. Not sure if this would have major implications though, so this is just a nit.

File: tests/strategies/test_strategy_load_hours.py [hunk 1]
```diff
@@ -601,3 +602,36 @@ def test_set_energy_measurement_of_last_market(utils_mock, load_hours_strategy_t
...
+
+
+def test_event_offer_traded_calls_settlement_market_event_offer_traded(load_hours_fixture):
+    """Test if _settlement_market_strategy.event_offer_traded was called
+    although no spot market can be found by event_offer_traded."""
+    load_hours_fixture._settlement_market_strategy = Mock()
+    offer = Offer("oid", None, 1, 1, "seller")
+    trade = Trade('idt', None, offer, 'B', load_hours_fixture.owner.name)
+    load_hours_fixture.event_offer_traded(market_id="not existing", trade=trade)
+    load_hours_fixture._settlement_market_strategy.event_offer_traded.assert_called_once()
```
Comment (hannesdiedrich) [inline tests/strategies/test_strategy_load_hours.py:637]: These are the added unittests.

File: src/d3a/models/area/__init__.py [hunk 2]
```diff
@@ -402,7 +402,7 @@ def tick(self):
         updates cached myco matcher markets and match trades recommendations.
         """
         self._consume_commands_from_aggregator()
-
+        print()
```
Comment (fievelk) [inline src/d3a/models/area/__init__.py:405]:
```suggestion
```

File: src/d3a/models/strategy/load_hours.py [hunk 3]
```diff
@@ -384,8 +384,13 @@ def event_bid_traded(self, *, market_id, bid_trade):
 
         This method is triggered by the MarketEvent.BID_TRADED event.
         """
+        # settlement market event_bid_traded has to be triggered before the early return:
+
+        self._settlement_market_strategy.event_bid_traded(self, market_id, bid_trade)
+
         super().event_bid_traded(market_id=market_id, bid_trade=bid_trade)
         market = self.area.get_future_market_from_id(market_id)
+        print(market)
```
Comment (fievelk) [inline src/d3a/models/strategy/load_hours.py:393]:
```suggestion
```

File: src/d3a/models/strategy/load_hours.py [hunk 4]
```diff
@@ -384,8 +384,13 @@ def event_bid_traded(self, *, market_id, bid_trade):
 
         This method is triggered by the MarketEvent.BID_TRADED event.
         """
+        # settlement market event_bid_traded has to be triggered before the early return:
+
```
Comment (fievelk) [inline src/d3a/models/strategy/load_hours.py:388]: Nit:
```suggestion
```

File: tests/strategies/external/test_init.py [hunk 5]
```diff
@@ -244,7 +244,8 @@ def test_dispatch_event_trade_to_external_aggregator(self, strategy, offer_bid):
                           "test_area", "parent_area", fee_price=0.23, buyer_id=self.area.uuid,
                           seller_id=self.parent.uuid)
 
-        strategy.event_trade(market_id="test_market", trade=trade)
+        strategy.event_offer_traded(market_id="test_market", trade=trade)
+        print(strategy.redis.aggregator.add_batch_trade_event.call_args_list)
```
[the remark this answers is not in the document]
Comment (hannesdiedrich) [reply tests/strategies/external/test_init.py:248]: Sorry, forgot to clean up apparently. Thanks for finding!

**Pull request merged**